src/sound/Sound.cpp

```diff
@@ -588,125 +195,107 @@ namespace Sound {
 		std::map<std::string, Sample> m_loadedSounds;
 	};
 
-	std::vector<std::string> s_audioDeviceNames = {};
+	std::vector<std::string_view> GetAvailableBackends()
+	{
+		std::vector<std::string_view> backends{ SDLBackendName };
+#ifdef PI_BUILD_WITH_OPENAL
+		backends.push_back(OpenALBackendName);
+#endif
+		return backends;
+	}
 
-	bool Init(bool automaticallyOpenDevice)
+	std::string_view GetBackend()
 	{
-		PROFILE_SCOPED()
-		if (m_audioDevice) {
-			DestroyAllEvents();
-			return true;
+		if (dynamic_cast<SdlAudioBackend *>(m_backend) != nullptr) {
+			return SDLBackendName;
+		} else if (dynamic_cast<AlAudioBackend *>(m_backend) != nullptr) {
+			return OpenALBackendName;
 		}
+		return "Unknown Sound Backend";
+	}
 
-		if (SDL_Init(SDL_INIT_AUDIO) == -1) {
-			Output("Could not initialize SDL Audio: %s.\n", SDL_GetError());
-			return false;
+	bool Init(std::string_view backend)
+	{
+		PROFILE_SCOPED()
+		if (m_backend != nullptr) {
+			if (backend == GetBackend()) {
+				DestroyAllEvents();
+				return true;
+			} else {
+				Uninit();
+			}
 		}
 
-		// load all the wretched effects
-		Pi::GetApp()->GetAsyncStartupQueue()->Order(new LoadSoundJob("sounds", false));
-
-		//I'd rather do this in MusicPlayer and store in a different map too, this will do for now
-		Pi::GetApp()->GetAsyncStartupQueue()->Order(new LoadSoundJob("music", true));
-
-		UpdateAudioDevices();
-
-		// If we're going to manually pick a device later, don't open a default one now.
-		if (!automaticallyOpenDevice) {
-			return true;
+		if (backend.empty()) {
+#ifdef PI_BUILD_WITH_OPENAL
+			backend = OpenALBackendName;
+#else
+			backend = SDLBackendName;
+#endif
```

Comment:
Do we want to default to OpenAL or SDL?

Usually you don't want to change a players' experience automatically..

Reply:
Sure; I wasn't aware that this was replacement work so I agree with your reasoning here now. Closing.